Discovering and learning everything there is to know about R packages using r-universe (#490)
ropensci/roweb3: content/blog/2023-02-24-runiverse-discovering-es/index.md

{{< figureforgallery src="../runiverse-discovering/package1.png" alt="Screenshot of r-universe package homepage" >}}

Estas páginas de paquetes se generan automáticamente y proporcionan información detallada sobre cada proyecto, así como toda la información y recursos para instalar y probar rápidamente el paquete localmente en R.

### Información del paquete

La sección inicial de la página de inicio muestra datos descriptivos y enlaces a recursos de este paquete, como los archivos fuente/binarios, el manual del paquete y el archivo NEWS. Más abajo mostramos las instrucciones de instalación para instalar el paquete en R. La instalación es rápida y sencilla porque proporcionamos binarios precompilados para Windows y MacOS, por lo que no se necesitan herramientas ni configuraciones especiales en la máquina del usuario.

{{< gallery >}}
{{< figureforgallery src="../runiverse-discovering/package2.png" alt="Screenshot of r-universe package homepage with annotations." >}}

> {{< figureforgallery src="../runiverse-discovering/package21.png" alt="Screenshot of r-universe package badges and statistics." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package21.png" alt="Ejemplo de una página mostrando las insignias y estadísticas de un paquete en r-universe." >}}
```

> {{< figureforgallery src="../runiverse-discovering/package4.png" alt="Screenshot of r-universe package: how to cite the package." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package4.png" alt="Ejemplo de como se presenta la información de como citar el paquete en r-universe." >}}
```

{{< /gallery >}}

> Aquí también puedes encontrar más información derivada por el sistema de construcción, como sobre qué bibliotecas del sistema construye el paquete, temas y si el paquete está en CRAN. Las etiquetas de debajo muestran estadísticas sobre el paquete, como dependencias, dependencias inversas y estrellas de GitHub. Haciendo clic en la etiqueta "exportaciones" se muestran todas las funciones y conjuntos de datos exportados por el paquete, que enlazan con las secciones respectivas de la página del manual. Si se especifica, también mostramos cómo debe citarse el paquete en los artículos.

Reviewer: ```suggestion
Aquí también puedes encontrar más información derivada por el sistema de construcción, como sobre qué bibliotecas del sistema construye el paquete, temas y si el paquete está en CRAN. Las etiquetas debajo muestran estadísticas sobre el paquete, como dependencias, dependencias inversas y estrellas de GitHub. Haciendo clic en la etiqueta "exports" se muestran todas las funciones y conjuntos de datos exportados por el paquete, que enlazan con las secciones respectivas de la página del manual. Si se especifica, también mostramos cómo debe citarse el paquete en los artículos.
```

### Toda la documentación

Para que puedas empezar a utilizar los paquetes lo antes posible, r-universe ofrece toda la documentación disponible y la presenta de la forma más clara y concisa posible en la página de inicio.

> La mayoría de los paquetes de R tienen 3 formas de documentación: un archivo léeme, un manual de referencia (páginas de ayuda) y una serie de artículos (viñetas). Estos artículos suelen ser un buen punto de partida, y aparecen en la página principal del paquete, justo debajo de los descriptivos. Desde aquí puedes ver tanto el documento html/pdf renderizado, como el archivo fuente rmarkdown utilizado para (re)producir el artículo.

Reviewer: ```suggestion
La mayoría de los paquetes de R tienen 3 formas de documentación: un archivo léeme (README), un manual de referencia (páginas de ayuda) y una serie de artículos (viñetas). Estos artículos suelen ser un buen punto de partida, y aparecen en la página principal del paquete, justo debajo de los descriptivos. Desde aquí puedes ver tanto el documento html/pdf renderizado, como el archivo fuente rmarkdown utilizado para (re)producir el artículo.
```

{{< gallery >}}

> {{< figureforgallery src="../runiverse-discovering/package3.png" alt="Screenshot of r-universe package, listing included articles." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package3.png" alt="Ejemplo del universo de un paquete listando todos los artículos incluidos." >}}
```

> {{< figureforgallery src="../runiverse-discovering/package5.png" alt="Screenshot of r-universe package showing example readme content." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package5.png" alt="Ejemplo del contenido de un archivo README en el univrso de un paquete." >}}
```

> {{< figureforgallery src="../runiverse-discovering/package6.png" alt="Screenshot of r-universe package showing reference manual index." >}}

Author: hmm I think I used the latest version

> {{< figureforgallery src="../runiverse-discovering/package6.png" alt="Screenshot of r-universe package showing reference manual index." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package6.png" alt="Ejemplo de la referencia al índice de un manual de un paquete en r-universe." >}}
```

{{< /gallery >}}

> Más abajo puedes ver el contenido del "Léame" del paquete. Hoy en día, la mayoría de los paquetes incluyen un archivo README.md que se puede convertir en HTML.

Reviewer: ```suggestion
Más abajo puedes ver el contenido del "README" del paquete. Hoy en día, la mayoría de los paquetes incluyen un archivo README.md que se puede convertir en HTML.
```

> Después del readme sigue una sección llamada "manual de referencia" que contiene una tabla con todas las páginas de ayuda. Éstas enlazan con los respectivos capítulos del manual de referencia html[^2], que proporciona la documentación más extensa de cada función del paquete, incluyendo descripciones, parámetros, ejemplos, etc.

Reviewer: ```suggestion
Después del README sigue una sección llamada "manual de referencia" que contiene una tabla con todas las páginas de ayuda. Éstas enlazan con los respectivos capítulos del manual de referencia html[^2], que proporciona la documentación más extensa de cada función del paquete, incluyendo descripciones, parámetros, ejemplos, etc.
```

> {{< figureforgallery src="../runiverse-discovering/refman.png" alt="Screenshot of a chapter in the HTML reference manual for a package, showing descriptions, parameters, and examples for a function." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/refman.png" alt="Ejemplo de un capítulo en en el manual de referencia en HTML de un paquete, mostrando descripciones, parámetros y ejemplos de una función." >}}
```

### Actividad de uso y desarrollo

Además de familiarizarte con lo que hace el paquete, puede que te interese saber quién está trabajando en él, quién más lo utiliza y si sigue desarrollándose activamente.

> La sección "desarrollo y colaboradores" muestra un gráfico de barras con el número de commits por semana durante el último año, y quiénes son los principales colaboradores. Por último, la sección "uso por otros paquetes" muestra otros paquetes de r-universo que dependen de este paquete, agrupados por propietario. Esto te da una idea de lo activo que es el proyecto y de quién está implicado.

Reviewer: ```suggestion
La sección ""development and contributors" (desarrollo y colaboraciones) muestra un gráfico de barras con el número de commits por semana durante el último año, y quiénes son los y las principales colaboradores y colaboradoras. Por último, la sección "usage by other packages" (uso por otros paquetes) muestra otros paquetes de r-universo que dependen de este paquete, agrupados por propietario/a. Esto te da una idea de lo activo que es el proyecto y de quién está participando.
```

> {{< figureforgallery src="../runiverse-discovering/package7.png" alt="Screenshot of recent development activity and users of a package." >}}

Reviewer: ```suggestion
{{< figureforgallery src="../runiverse-discovering/package7.png" alt="Ejemplo de actividad reciente de desarrollo y personas que usan el paquete." >}}
```

## Conclusión

> Este post ha explicado diferentes formas en las que puedes utilizar [r-universo](https://r-universe.dev) para explorar los paquetes de R. La página [motor de búsqueda](https://r-universe.dev/search) te permite encontrar paquetes de cualquier parte del ecosistema que coincidan con determinadas palabras clave o criterios específicos. Es un buen lugar para informarse y hacerse una idea de lo que hay disponible en un dominio determinado.

Reviewer: ```suggestion
Este post ha explicado diferentes formas en las que puedes utilizar [r-universe](https://r-universe.dev) para explorar los paquetes de R. La página [motor de búsqueda](https://r-universe.dev/search) te permite encontrar paquetes de cualquier parte del ecosistema que coincidan con determinadas palabras clave o criterios específicos. Es un buen lugar para informarse y hacerse una idea de lo que hay disponible en un dominio determinado.
```

> Una vez que te familiarices con el ecosistema, puede que te interese hacer zoom en un conjunto específico de paquetes de un determinado grupo o investigador. La estructura de dominios de r-universo te permite explorar paquetes y artículos por organización, y descubrir trabajos relacionados con mantenedores, colaboradores o usuarios similares.

Reviewer: ```suggestion
Una vez que te familiarices con el ecosistema, puede que te interese hacer zoom en un conjunto específico de paquetes de un determinado grupo o investigador/a. La estructura de dominios de r-universe te permite explorar paquetes y artículos por organización, y descubrir trabajos relacionados con las personas que los mantienen, colaboran o los usan.
```

Por último, para cada paquete R, r-universe proporciona una hermosa página de inicio que contiene abundante información en: `https://{owner}.r-universe.dev/{package}`. Esta página muestra exactamente de qué trata un paquete, quién lo está desarrollando, el estado del proyecto, y también proporciona toda la documentación y las instrucciones de instalación, muy bien representadas, para empezar a utilizar el paquete inmediatamente, sin complicaciones.

[^1]: Más información sobre esta clasificación: [¿cómo se calcula la puntuación de la clasificación por paquetes?](https://github.com/r-universe-org/help#how-is-the-package-rank-score-calculated)

> [^2]: Entrada en el blog: [postdoc 1.0: manuales de paquetes HTML mínimos y despejados](/blog/2022/11/29/postdoc-docs/).

Reviewer: ```suggestion
[^2]: Entrada en el blog: [postdoc 1.0: manuales de paquetes HTML mínimos y claros](/blog/2022/11/29/postdoc-docs/).
```
